core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,555 @@
-
-const mobileNavToggelbutton = '.e2e-test-mobile-options';
-const mobileChangesDropdown = '.e2e-test-mobile-changes-dropdown';
-const mobileSaveChangesButton =
-  'button.e2e-test-save-changes-for-small-screens';
+const settingsTab = 'a.e2e-test-exploration-settings-tab';
+const addTitleBar = 'input#explorationTitle';
+const addGoalInputBox = '.e2e-test-exploration-objective-input';
+const categoryDropdown = 'mat-select.e2e-test-exploration-category-dropdown';
+const languageUpdateDropdown =
+  'mat-select.e2e-test-exploration-language-select';
+const addTagsInputBox = 'input.e2e-test-chip-list-tags';
+const previewSummaryButton = 'button.e2e-test-open-preview-summary-modal';
+const dismissPreviewButton = 'button.e2e-test-close-preview-summary-modal';
+const textToSpeechToggle = 'label.e2e-test-on-off-switch';
+const feedbackToggle = 'label.e2e-test-enable-fallbacks';
+
+const editRoleButton = '.e2e-test-edit-roles';
+const addUsernameInputBox = '#newMemberUsername';
+const addRoleDropdown = 'mat-select.e2e-test-role-select';
+const collaboratorRoleOption = 'Collaborator (can make changes)';
+const playtesterRoleOption = 'Playtester (can give feedback)';
+const saveRoleButton = 'button.e2e-test-save-role';
+const desktopSaveDraftButton = 'button.e2e-test-save-changes';
+const confirmSaveDraftButton = '.e2e-test-save-draft-button';
+const desktopPublishButton = 'button.e2e-test-publish-exploration';
+const discardDraftDropdown = 'button.e2e-test-save-discard-toggle';
+const desktopDiscardDraftButton = 'a.e2e-test-discard-changes';
+const confirmDiscardButton = 'button.e2e-test-confirm-discard-changes';
+const confirmPublishButton = 'button.e2e-test-confirm-publish';
+const commitMessage = 'textarea.e2e-test-commit-message-input';
+const closePublishedPopUpButton = 'button.e2e-test-share-publish-close';
+
+/**
+ * For mobile.
+ */
+const navBarOpenerIcon = 'i.e2e-test-mobile-options';
+const optionsDropdown = 'div.e2e-test-mobile-options-dropdown';
+const mobileSettingsBar = 'li.e2e-test-mobile-settings-button';
+const basicSettingsDropdown = 'h3.e2e-test-settings-container';
+const feedbackSettingsDropdown = 'h3.e2e-test-feedback-settings-container';
+const permissionSettingsDropdown = 'h3.e2e-test-permission-settings-container';
+const voiceArtistSettingsDropdown =
+  'h3.e2e-test-voice-artists-settings-container';
+const rolesSettingsDropdown = 'h3.e2e-test-roles-settings-container';
+const advanceSettingsDropdown = 'h3.e2e-test-advanced-settings-container';
+const explorationControlsSettingsDropdown =
+  'h3.e2e-test-controls-bar-settings-container';
+const mobileSaveDraftButton = 'button.e2e-test-save-changes-for-small-screens';
 const mobilePublishButton = 'button.e2e-test-mobile-publish-button';
+const publishButtonDropdown = 'div.e2e-test-mobile-changes-dropdown';
+const mobileDiscardButton = 'div.e2e-test-mobile-exploration-discard-tab';
+
+let explorationUrlAfterPublished = '';
```

Comment:
Remove this line since we're not allowing mutable state.

Reply:
done